modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -569,6 +570,28 @@ func (h *Handler) proxyLoopIteration(r *http.Request, origReq *http.Request, w h
 	// or satisfactorily represented in a URL
 	caddyhttp.SetVar(r.Context(), dialInfoVarKey, dialInfo)
 
+	var remoteAddr string
+	if xff := r.Header.Get("X-Forwarded-For"); xff != "" {
```

Comment:
I think we should wait for [pr](https://github.com/caddyserver/caddy/pull/5104) to be merged. Parsing xff is not good.

Reply:
Good point. I'll keep that in mind as I finish that PR.

Reply:
@francislavoie Please let me know once merged and I'll rebase/ adapt.

Reply:
Alright, rebased & adjusted. @gucki @WeidiDeng would you mind confirming that it still works as expected?

Reply:
It still works according to my test.